1 init preprocessing logic

liushuang-118 opened the pull request:
- Factor out the pre-processing logic from the training pipeline.
- Fetch relevant dependencies through a package manager, e.g., by creating a separate requirements.txt.
- A workflow is used to automatically release the library in a package registry that matches the language.
- The library is versioned automatically, e.g., by picking-up on the corresponding Git version tag.

tfrvanhoorn commented:
Great job. I have a few remarks:

- It might be a good idea to add a CI workflow (maybe not the highest priority at the moment)
- Automatic versioning should eventually have pre-versioning as well as stable releases. I managed to make that work in my workflows. You can take a look at the delivery and deployment workflows in the app-frontend repository for that. I use a GitHub action that was published by someone else, which makes it a lot less complex. If you have any questions about that, feel free to ask.

RaoulSaad reviewed (commented):
The library works great!

The only thing that might need to be changed is the versioning workflow to add the automatic versioning (with the pre-releases) that still needs to be implemented. @tfrvanhoorn has a good reference in the app repository

liushuang-118 commented:
Thanks! I agree with your suggestions. I've added the integration.yml workflow and updated the deployment workflow to include automatic versioning with pre-releases. I also referenced  @tfrvanhoorn’s repository for guidance.

liushuang-118 commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,43 @@
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
```
This is the code I wrote earlier, I have added the version in the latest yml files.

RaoulSaad reviewed (approved):
Looks all good
